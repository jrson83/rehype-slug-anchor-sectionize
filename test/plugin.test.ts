import type { RehypeSlugAnchorSectionizeOptions } from '../src'

import { expect, it } from 'vitest'

import { readFile } from 'node:fs/promises'
import { resolve } from 'node:path'

import rehypeStringify from 'rehype-stringify'
import remarkParse from 'remark-parse'
import remarkRehype from 'remark-rehype'
import { unified } from 'unified'
import rehypeSlugSectionizeAnchorize from '../src'

const cwd = process.cwd()
const path = (file: string) => resolve(cwd, 'test', 'markdown', file)

function createProcessor(options?: RehypeSlugAnchorSectionizeOptions) {
  const processor = unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(rehypeSlugSectionizeAnchorize, options)
    .use(rehypeStringify)

  return processor
}

it('should ...', async () => {
  const { value } = await createProcessor().process(
    await readFile(path('markdown.md'), { encoding: 'utf8' })
  )

  expect(value).toMatchSnapshot()
})
