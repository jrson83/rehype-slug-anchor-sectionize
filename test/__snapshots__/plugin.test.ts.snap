// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should ... 1`] = `
"<section id=\\"heading-21-section\\"><h2><a href=\\"#heading-21\\" class=\\"heading-anchor\\"><button type=\\"button\\" class=\\"heading-anchor__btn\\">#</button></a><span role=\\"heading\\" aria-level=\\"2\\">Heading 2.1</span></h2>
<p>Paragraph 1</p>
</section><section id=\\"heading-22-section\\"><h2><a href=\\"#heading-22\\" class=\\"heading-anchor\\"><button type=\\"button\\" class=\\"heading-anchor__btn\\">#</button></a><span role=\\"heading\\" aria-level=\\"2\\">Heading 2.2</span></h2>
<p>Paragraph 2</p>
<section id=\\"heading-31-section\\"><h3><a href=\\"#heading-31\\" class=\\"heading-anchor\\"><button type=\\"button\\" class=\\"heading-anchor__btn\\">#</button></a><span role=\\"heading\\" aria-level=\\"3\\">Heading 3.1</span></h3>
<p>Paragraph 3</p>
</section></section><section id=\\"heading-23-section\\"><h2><a href=\\"#heading-23\\" class=\\"heading-anchor\\"><button type=\\"button\\" class=\\"heading-anchor__btn\\">#</button></a><span role=\\"heading\\" aria-level=\\"2\\">Heading 2.3</span></h2>
<p>Paragraph 4</p></section>"
`;
